plugin/keyring_vault/vault_base64.h

```diff
@@ -0,0 +1,23 @@
+#ifndef MYSQL_VAULT_BASE64_H
+#define MYSQL_VAULT_BASE64_H
+
+#include <my_global.h>
+#include "vault_memory.h"
+
+namespace keyring
+{
+  class Vault_base64
+  {
+  public :
+    enum Base64Format
+    {
+      SINGLE_LINE,
+      MULTI_LINE
+    };
+    static bool encode(const void *src, size_t src_len, Secure_string *encoded, Base64Format format);
```

Comment:
I am not even sure it did make sense to add SINGLE_LINE/MUTLTILINE parameters to decode function in the first place. This functions are only to be used inside keyring_vault and all this gave me is one more extra parameter. Now everywhere I can Vault_base64::encode with SINGLE_LINE, never with multiple. Those are not general function - base64.h is for that. If enything I would remove Base64Format at all - it is just confusing imho.